web/modules/custom/dbcdk_community/src/Plugin/Block/FlaggedContentDetails.php

```diff
@@ -0,0 +1,189 @@
+    $this->logger = $logger;
+    $this->flaggableContentRepository = $repository;
+    $this->profileApi = $profile_api;
+    $this->urlGenerator = $url_generator;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    /* @var \Drupal\Core\Logger\LoggerChannelFactory $logger_factory */
+    $logger_factory = $container->get('logger.factory');
+
+    // Setup url generation for flaggable content.
+    /* @var \Drupal\Core\Config\Config $config */
+    $config = $container->get('config.factory')->get('dbcdk_community.settings');
+    $base_url = $config->get('community_site_url');
+    $url_generator = new ModelUrlGenerator();
+    $url_generator->registerClassGenerator(
+      'DBCDK\CommunityServices\Model\Post',
+      new PropertyUrlGenerator(
+        $base_url . $config->get('community_site_post_url_pattern')
+      )
+    );
+    $url_generator->registerClassGenerator(
+      'DBCDK\CommunityServices\Model\Comment',
+      new PropertyUrlGenerator(
+        $base_url . $config->get('community_site_comment_url_pattern')
+      )
+    );
+
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $logger_factory->get('DBCDK Community Service'),
+      $container->get('dbcdk_community.content.flaggable_content_repository'),
+      $container->get('dbcdk_community.api.profile'),
+      $url_generator
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function build() {
+    $flag_id = $this->getContextValue('flag_id');
+
+    try {
+      $content = $this->flaggableContentRepository->getContentById($flag_id);
+      $profile = $this->profileApi->profileFindById($content->getCreatorId());
+
+      $profile_link = Link::createFromRoute(
+        $profile->getUsername(),
+        'page_manager.page_view_dbcdk_community_profile',
+        ['username' => $profile->getUsername()]
+      );
+
+      $url = $this->urlGenerator->generate($content->getObject());
+      $content_link = Link::fromTextAndUrl($url, Url::fromUri($url));
+
+      $rows = [
+        [$this->t('Content'), $content->getContent()],
+        [$this->t('Created on'), $content->getTimeCreated()->format('H:i d-m-Y')],
+        [$this->t('Author'), $profile_link],
+        [$this->t('View on biblo.dk'), $content_link],
+      ];
+    }
+    catch (ApiException $e) {
+      // If an error occours then log it and display an empty table.
+      $this->logger->error($e);
+      $rows = [];
+    }
+
+    $table = [
+      '#theme' => 'table',
+      '#header' => [
+        $this->t('Subject'),
+        $this->t('Information'),
+      ],
+      '#rows' => $rows,
+      '#empty' => $this->t('No information found for content related to flag "%flag_id".', [
+        '%flag_id' => $flag_id,
+      ]),
+      '#cache' => [
+        'max-age' => 0,
+      ],
+    ];
+
+    return [
+      'table' => $table,
```

Comment:
Did you intend for the table to be more dynamical since it's defined in a variable instead of here? You could create a "build" variable with a table key (`$build['table'] = ...`). By doing this, you'll be able to easily add more to the render array later on, if this was your intentions from the beginning?